jobs/src/main/scala/dcos/metronome/utils/state/ZKStore.scala

```diff
@@ -0,0 +1,163 @@
+package dcos.metronome.utils.state
+
+import java.util.UUID
+
+import com.fasterxml.uuid.impl.UUIDUtil
+import com.google.protobuf.{ ByteString, InvalidProtocolBufferException }
+import com.twitter.util.{ Future => TWFuture }
+import com.twitter.zk.{ ZNode, ZkClient }
+import mesosphere.marathon.{ Protos, StoreCommandFailedException }
+
+import org.apache.zookeeper.KeeperException
+import org.apache.zookeeper.KeeperException.{ NoNodeException, NodeExistsException }
+import org.slf4j.LoggerFactory
+import ZKStore._
+
+import scala.concurrent.{ ExecutionContext, Future, Promise }
+
+case class CompressionConf(enabled: Boolean, sizeLimit: Long)
+
+class ZKStore(val client: ZkClient, root: ZNode, compressionConf: CompressionConf) extends PersistentStore
```

Comment:
Out of curiosity, why are we copying what Marathon already provides?

Reply:
this is what marathon 1.3 provided, but it's no longer part of 1.5